src/lib/db/adapters/dbAdapter.js

```diff
@@ -117,15 +117,41 @@ const upsertReport = async (reportData) => {
 const listReports = async (queryParams) => {
   const conn = await getConnection()
   const MyModel = conn.model('reports', mongoDBWrapper.models.reports)
-  // by default is taking the reports from the last 30 days
-  const query = {
-    'runtimeInformation.completedTime': {
-      $gte: queryParams?.gte ? new Date(queryParams.gte) : new Date(Date.now() - (30 * 24 * 60 * 60 * 1000)),
-      $lt: queryParams?.lt ? new Date(queryParams.lt) : new Date()
+  const query = {}
+  if (queryParams?.filterDateRangeStart || queryParams?.filterDateRangeEnd) {
+    query['runtimeInformation.completedTime'] = {}
+    if (queryParams?.filterDateRangeStart) {
+      query['runtimeInformation.completedTime'].$gte = new Date(queryParams?.filterDateRangeStart)
     }
+    if (queryParams?.filterDateRangeEnd) {
+      query['runtimeInformation.completedTime'].$lte = new Date(queryParams?.filterDateRangeEnd)
+    }
+  }
+
+  if (queryParams?.filterStatus) {
```

Comment:
This seems like an Unnecessary if statement?

Reply:
Thanks, removed it.